### ISSUE TITLE
Functions integer-to-string and string-to-integer with radix

### ISSUE THREAD

**Issue by michaelhkay** [qt4cg/qt4tests#27]
I propose (in response to a suggestion from Joel Kalvesmaki on the XML Slack) two functions

```
fn:integer-to-string($value as xs:integer, $radix as xs:integer := 10) as xs:string

fn:string-to-integer($value as xs:string, $radix as xs:integer := 10) as xs:integer
```

that produce/accept string representations of signed integers in a base other than 10. Proposed range for $radix is 2 to 32.

Note:  I considered extending `fn:format-integer()`, but the result looked clumsy.

**Comment by ndw** [comment]
Also in that thread is a mention of the EXPath functions and I wonder if we need to consider additional function libraries. On the one hand, putting everything in `fn` makes life simpler for users, but it increases the size of the "core" of XPath. Maybe we don't care, but maybe we do.

**Comment by michaelhkay** [comment]
I looked at the EXPath binary module and the interesting thing is, it doesn't actually have this capability. You can parse hex, binary or octal by using bin:hex (etc) to get an octet sequence and then unpacking an integer from that, but it's rather convoluted, and doesn't handle other radices; in the other direction (int-to-string) I can only see how to do decimal and hex.

**Comment by ndw** [comment]
Right. Sorry. I wasn't suggesting that this already existed in EXPath, but it's at least in the neighborhood of some of the other functions in EXPath. It seems to me that it could have been proposed as an EXPath extension or a core function, and I don't see any obvious rationale for which to choose.

I think the proposal for base64 encoding, which does exist in EXPath, is of a similar kind of utility. (If you want to make a `data:` URI to inline an image to avoid another round-trip to the server, you have to base64 encode it.)

**Comment by michaelhkay** [comment]
Base64 is well covered by the EXPath Binary module, I think, and I don't see any need for further facilities there.

**Comment by michaelhkay** [comment]
Oh dear, I put this issue under qt4tests rather than qt4specs. Not sure if it can be moved?
